### PR TITLE
Update to latest k8s.io/utils to pick up changes

### PR DESCRIPTION
**Pull request by dims**
Pick up updates to k8s.io/utils/pointer/pointer.go and go.mod as well

/kind cleanup

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by liggitt** [comment]
deps is complaining about pinned versions... I'm surprised anything other than k8s.io/utils changed

looks like the afero update pulled in a lot of stuff... opened https://github.com/kubernetes/utils/pull/261 to prune out test-only deps

**Comment by dims** [comment]
@liggitt updated with latest k8s.io/util after https://github.com/kubernetes/utils/pull/261 merged. Looks way better now.

thanks!

File: vendor/modules.txt [hunk 1]
```diff
@@ -2718,7 +2716,6 @@ sigs.k8s.io/yaml
 # github.com/sergi/go-diff => github.com/sergi/go-diff v1.1.0
 # github.com/sirupsen/logrus => github.com/sirupsen/logrus v1.8.1
 # github.com/soheilhy/cmux => github.com/soheilhy/cmux v0.1.5
-# github.com/spf13/afero => github.com/spf13/afero v1.6.0
```
Comment (liggitt) [inline vendor/modules.txt:2721]: 🎉

**Comment by liggitt** [comment]
```diff
 {
-	"directDependencies": 217,
-	"transitiveDependencies": 258,
-	"totalDependencies": 311,
+	"directDependencies": 216,
+	"transitiveDependencies": 255,
+	"totalDependencies": 308,
 	"maxDepthOfDependencies": 30
 }
```

**Comment by liggitt** [comment]
/retest

weird integration flake... I've seen some timeout / server errors hitting that recently :-/

**Comment by liggitt** [comment]
fwiw, this is our progression over the last several minor versions:

```
File: 1.19.json
{
    "directDependencies": 168,
    "transitiveDependencies": 265,
    "totalDependencies": 333,
    "maxDepthOfDependencies": 25
}
File: 1.20.json
{
    "directDependencies": 167,
    "transitiveDependencies": 297,
    "totalDependencies": 364,
    "maxDepthOfDependencies": 31
}
File: 1.21.json
{
    "directDependencies": 163,
    "transitiveDependencies": 310,
    "totalDependencies": 369,
    "maxDepthOfDependencies": 32
}
File: 1.22.json
{
    "directDependencies": 163,
    "transitiveDependencies": 307,
    "totalDependencies": 364,
    "maxDepthOfDependencies": 31
}
File: 1.23.json
{
    "directDependencies": 163,
    "transitiveDependencies": 307,
    "totalDependencies": 367,
    "maxDepthOfDependencies": 30
}
File: 1.24.json
{
    "directDependencies": 163,
    "transitiveDependencies": 262,
    "totalDependencies": 323,
    "maxDepthOfDependencies": 29
}
File: 1.25.json
{
    "directDependencies": 228,
    "transitiveDependencies": 262,
    "totalDependencies": 322,
    "maxDepthOfDependencies": 30
}
File: 1.26.json
{
    "directDependencies": 216,
    "transitiveDependencies": 255,
    "totalDependencies": 308,
    "maxDepthOfDependencies": 30
}
```

some of the trajectories are a little misleading because of bumps to dependencies that added go.mod files (which exposed way more transitives to `go mod graph`) or to go 1.17+ (which made way more things appear to be direct dependencies), but the overall totalDependencies trajectory since 1.21 is encouraging

**Comment by dims** [comment]
We've gotta keep at it! thanks for pulling the data together @liggitt

**Comment by chenfanlinux** [comment]
/

**Pull request merged**